velox/common/memory/Memory.h

```diff
@@ -124,6 +130,10 @@ class MemoryPool : public std::enable_shared_from_this<MemoryPool> {
   /// Returns the name of this memory pool.
   virtual const std::string& name() const;
 
+  /// Returns the shared pointer to the parent pool. The root memory pool has
+  /// no parent.
+  virtual std::shared_ptr<MemoryPool> parent() const;
```

Comment:
Why do we return shared_ptr and not a raw pointer?

Reply:
Good question. Then we can't guarantee the parent pointer is always valid. We shall comment say the parent raw pointer is safe to access while user hold the reference to this child memory pool. Updated.